fetch_github_data.sh

```diff
@@ -0,0 +1,114 @@
+
+#TODO: we don't use this data anywhere, maybe don't fetch it?
+#echo 'https://api.github.com/repos/'${REPO}'/comments?per_page=100' | GITHUB_TOKEN=$GITHUB_TOKEN ALLCOMMENTS="" ${DIR}/fetch-comments.sh | gzip -c > ${DATADIR}/commit-comments.gz
+
+github_fetch_time=$(( $(date +%s) - ${starttime}))
+echo "done fetching pull request and issue data in ${github_fetch_time}s."
+echo
+printf "github_fetch_time\t%f\n" ${github_fetch_time} >> ${DATADIR}/times.tsv
+
+echo "checking with GitHub how a sample of commits relate to pulls..."
+starttime=$(date +%s)
+cat ${DATADIR}/commitdates |\
+    gawk -vrepo=${REPO} -vearliest=$(date -d "${EARLIEST_DATE}" +%s) -F\\t '$2>=earliest {printf("https://api.github.com/repos/%s/commits/%s/pulls\n", repo, $1)}' |\
+    sort -R | tail -n${COMMIT_PR_SAMPLESIZE} |\
+    GITHUB_TOKEN=$GITHUB_TOKEN HEADER_ACCEPT="application/vnd.github.groot-preview+json" PREFIX_URL=true SILENT=true ${DIR}/fetch-comments.sh | pv -l -s200 \
+    > ${DATADIR}/commit_pulls
+github_commit_pull_time=$(( $(date +%s) - ${starttime}))
+echo "done pulling commit pull request info in ${github_commit_pull_time}s."
+echo
+printf "github_commit_pull_time\t%f\n" ${github_commit_pull_time} >> ${DATADIR}/times.tsv
+
+if ! (cat ${DATADIR}/pulls.gz | zcat | head -n1 || true) | grep ',    "number": [0-9]*' > /dev/null; then
+    echo "there are no PRs in this repository"
+    echo
+
+    #just leave the sample files empty
+    touch ${DATADIR}/pr_sample
+    touch ${DATADIR}/pr_sample_pulls
+    touch ${DATADIR}/pr_sample_commits
+else
+    #get a sample of pulls to prep some additional data
+    echo "preparing additional per-PR data for a ${PR_SAMPLESIZE} PR sample of PRs..."
+
+    starttime=$(date +%s)
+    echo "drawing the sample..."
+    pv ${DATADIR}/pulls.gz | zcat | grep -o ',    "number": [0-9]*' | sed -e 's/.* \([0-9]*\)$/\1/' | sort -R | tail -n${PR_SAMPLESIZE} | LC_ALL=C sort > ${DATADIR}/pr_sample
+
+    echo "fetching full pull objects for PR sample (this might take a while)..."
+    cat ${DATADIR}/pr_sample | gawk '{printf("https://api.github.com/repos/'${REPO}'/pulls/%d?per_page=100\n", $1)}' | GITHUB_TOKEN=$GITHUB_TOKEN ALLCOMMENTS="" SILENT=true ${DIR}/fetch-comments.sh | pv -l -s$(cat ${DATADIR}/pr_sample | wc -l) > ${DATADIR}/pr_sample_pulls
+
+    echo "fetch commit list for PR sample (this might take a while)..."
+cat ${DATADIR}/pr_sample | gawk '{printf("https://api.github.com/repos/'${REPO}'/pulls/%d/commits?per_page=100\n", $1)}' | GITHUB_TOKEN=$GITHUB_TOKEN ALLCOMMENTS="" SILENT=true PREFIX_URL=true ${DIR}/fetch-comments.sh | pv -l -s$(cat ${DATADIR}/pr_sample | wc -l) > ${DATADIR}/pr_sample_commits
+
+    github_pull_sample_time=$(( $(date +%s) - ${starttime}))
+    echo "done collecting sample PR data in ${github_pull_sample_time}s."
+    echo
+    printf "github_pull_sample_time\t%f\n" ${github_pull_sample_time} >> ${DATADIR}/times.tsv
+fi
+
+data_pull_time=$(( $(date +%s) - ${data_pull_start_time} ))
```

Comment:
I think you need to initialize `data_pull_start_time`

Reply:
See above.